.github/workflows/_build.yml

```diff
@@ -175,10 +174,76 @@ jobs:
       - name: Run Tests
         run: ./scripts/ci ${{ matrix.schemes }} test
 
+  Screenshots:
+    name: Screenshots tests
+    runs-on: macos-latest
+    needs: [Setup]
+    permissions:
+      pull-requests: write
+      contents: write
+    timeout-minutes: 20
+    continue-on-error: true
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Cache Bundler install Gems
+        uses: actions/cache@v4
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gems-${{ env.ImageVersion }}-${{ hashFiles('**/Gemfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-gems-${{ env.ImageVersion }}
+
+      - name: Cache Pods
+        uses: actions/cache@v4
+        with:
+          path: Example/Pods
+          key: ${{ runner.os }}-pods-${{ hashFiles('**/Podfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-pods-
+
+      - name: Bundle Install
+        run: bundle install --jobs 4 --retry 3
+
+      - name: Pod Install
+        run: bundle exec pod install
+        working-directory: Example
+
+      - name: Select Xcode 14
+        run: sudo xcode-select --switch /Applications/Xcode_14.2.app
+      
+      - name: Screenshot Tests
+        id: screenshotTests
+        run: |
+          ./scripts/record_screenshot_tests.sh
+
+          changedFiles=`git status --porcelain` && echo "CHANGED_FILES=${changedFiles//$'\n'/'%0A'}" >> $GITHUB_OUTPUT
+        continue-on-error: true
```

Comment:
is this needed because the step "fails" when snapshots change? what happens if there's a build error? we currently have a (slightly similar/related) issue on android where if that happens we delete all snapshots as none get generated. we don't have `continue-on-error` but the step didn't fail as it was missing the `set -e` at the top